### ISSUE TITLE
SMERF baking wrong results and missing js files in Webviewer

### ISSUE THREAD

**Issue by Yochengliu** [google-research/google-research#2097]
@duckworthd  Great work 👍 
After baking process on our own dataset, the render results are somewhat wrong, with many messy black regions (simple meeting room). But the original_render results are quite good. Any debug suggestions :) 
<img width="455" alt="image" src="https://github.com/google-research/google-research/assets/12179566/0506e672-8e6a-4e32-b1c0-4336235e2f5c">

The Webviewer can't work without all js files available (listed in **index.html**). When will these missing files be open-source? We are looking forward to experience real-time smooth and good rendering provided by your excellent SMERF.
![image](https://github.com/google-research/google-research/assets/12179566/17f00f6d-dce1-49d6-99b3-0f6fc41d0389)
![image](https://github.com/google-research/google-research/assets/12179566/6f586a3e-1259-48c0-8ccb-16703464d844)

Many thanks.

**Comment by duckworthd** [comment]
Hi @Yochengliu,

Thanks for pointing out the missing Javascript files. You can find them [here](https://github.com/smerf-3d/smerf-3d.github.io/tree/main/viewer). I'll update the repo here soon.

As for why baked renders look poor while original renders look good: this suggests that an insufficient number of views were used for baking. Unless configured otherwise, the set of all training views will be passed to the teacher's PropMLP to determine which voxels contain content for baking. I recommend verifying that [these `Config` parameters](https://github.com/google-research/google-research/blob/master/smerf/smerf/internal/configs.py#L136-L147) are set to their default values in `$CHECKPOINT_DIR/config.gin`.

If you have verified that these parameters are left to their default values and quality is still poor, I recommend loading the `alive_voxels.npy` file stored under `$CHECKPOINT_DIR/temp/sm_$SUBMODEL_ID/alive_voxels.npy` in Jupyter and visualizing the high-resolution voxel grid. It is likely that this grid is incomplete. 

NOTE: **The `temp/` directory is used as a cache** and must be manually deleted if any hyperparameters related to baking are changed! 

Best,
Daniel

**Comment by Yochengliu** [comment]
@duckworthd  Hi, thanks for your detailed suggestions, but the issue still exists. Below I upload debug information as much as possible. Please check them :) 

To remove the influence of customized datasets, I reproduce SMERF on the **Room** scene of mipNeRF 360, with default setting (only modify some print setting and batch size). You can check [**config.gin**](https://github.com/google-research/google-research/files/15349339/config.txt). ZipNeRF is trained by the this [repo](https://github.com/jonbarron/camp_zipnerf). Below the results are from my reproduction. There are the same issues (wrong depth and baking results) as on our own dataset.

### Issue 1: Wrong depth from SMERF
| ZipNeRF depth       | SMERF depth       |  RGB  |
|--------------|--------------|--------------|
| <img src="https://github.com/google-research/google-research/assets/12179566/50f07974-953d-413f-8503-40e4eee87077" width="256" height="256"> |  <img src="https://github.com/google-research/google-research/assets/12179566/2fcb7d48-9454-49bf-8bc7-7a3af96d86d7" width="256" height="256">  | <img src="https://github.com/google-research/google-research/assets/12179566/8e3b47c9-2a7c-4d26-8c27-82c2f89cd0a6" width="256" height="256"> |

ZipNeRF depth is obtained from **teacher_rendering['distance_median'][1]** in [train.py](https://github.com/google-research/google-research/blob/master/smerf/smerf/train.py#L277). You can see that SMERF has some regions with wrong depths. They seem to happen at dark regions. The situation becomes much worse on our own dataset, i.e., the depths at ribbonlike and regular black regions are almost all wrong (depth below). I speculate that this may be because some backgound color masking tricks in SMERF, but I didn't find the codes's location. Or is it because that I use the ZipNeRF without camp? Please give some suggestions :)
<img src="https://github.com/google-research/google-research/assets/12179566/55958da5-5ee5-47ac-b8ec-55d6c63b2a73" width="256" height="256">

### Issue 2: Incomplete baking results.
https://github.com/google-research/google-research/assets/12179566/d49503db-0ae1-4fee-9325-6dad156b3d02

You can see the **Room** baking results also have many black regions, just like the cases happen on our own dataset. I save training and baking log for your check. [train & bake.log](https://github.com/google-research/google-research/files/15349571/1.log) 
BTW, on our own dataset, I make sure that all images are used for baking and the config is with default setting as the config block you referred to. I also checked **alive_voxels.npy**, the proportion of alive voxels is only 2%, too few. Please give some suggestions :)

Best,
Yocheng

**Comment by Yochengliu** [comment]
@duckworthd  Hi, the baking issues are solved. The reason is that triplane_resolution and sparse_grid_resolution must meet a certain proportional relationship (4:1). Yet I set (1024, 384) on the above experiments.

I still have no idea why SMERF depths have some regular errors. Any suggestions :)

**Comment by duckworthd** [comment]
I'm glad to hear you found a solution! I'll add a check to ensure that `triplane_resolution` and `sparse_grid_resolution` are aligned.

**Comment by duckworthd** [comment]
The webviewer code has been updated and instructions have been added.

**Comment by Yochengliu** [comment]
@duckworthd   Many thanks.

What about "Wrong depth from SMERF" visualized above? Any comments?

Additionally, I found that the rendering results near the camera are quite blurred, i.e., the bottom third of the image, especially at characters‘ detail. I speculate the blur may be caused by multi-level hash encoding setting in InstantNGP. Do you have any specific suggestions :)

**Comment by duckworthd** [comment]
@sumanttyagi >> Yes, those 3 steps are correct. Please check out the research paper to build a mental model of how SMERF works and what's needed.
